### PR TITLE
remind: add support for dates in future

### PR DESCRIPTION
**Pull request by alanhuang122**
This pull request addresses issue #1148 by adding support for future dates, not just the next occurrence of the specified time.

### CODE REVIEW

**Comment by alanhuang122** [comment]
I'm unsatisfied with the code quality upon review. If this functionality is desired, I can make a new PR.

**Comment by dgw** [comment]
@alanhuang122 BTW, this functionality is desired. See #1148, which I've left open.

**Comment by alanhuang122** [comment]
I'll work on this ~eventually~.

**Comment by alanhuang122** [comment]
Pushed a cleaned up version of the code.
I think Flake8 is complaining about the fact that I'm using raw string for the input :|

File: sopel/modules/remind.py [hunk 1]
```diff
@@ -163,50 +163,103 @@ def remind(bot, trigger):
...
+        pytz_timezone = pytz.timezone(timezone)
+        now = pytz_timezone.localize(datetime.now())
+        at_time = datetime(year, month, day,
+                           int(hour), int(minute), int(second))
+        at_time = pytz_timezone.localize(at_time)
+        timediff = pytz_timezone.normalize(at_time) - now
     else:
         if tz and tz.upper() != 'UTC':
-            bot.reply("I don't have timezone support installed.")
+            bot.reply("I don't have timzeone support installed.")
```
Comment (dgw) [inline sopel/modules/remind.py]: This change introduces a spelling error that wasn't there before. Lol how? :joy_cat:

**Comment by dgw** [comment]
> and force-pushed to appease Travis. Twice.

Story of my life :joy_cat:

I'll look at this & give it a good shakedown over the weekend, hopefully. Thanks for reviving it!

File: sopel/modules/remind.py (hunk 1 above)
Comment (alanhuang122) [reply sopel/modules/remind.py]: I removed that code in my own copy then put it back in for the PR back then, evidently. How strange.

**Comment by alanhuang122** [comment]
Ack'd.
I do recall also finding some issue with timezones (probably the same) a few days ago, but haven't yet gotten around to porting the fix I (tried to?) make.

**Comment by alanhuang122** [comment]
Fixed and squashed (latest commit is preserved, for review).

**Comment by Exirel** [comment]
It would be much easier to review this with confidence if it has unit-tests. For that, I suggest to extract everything until `duration = timediff.seconds + 86400 * timediff.days` into its own function. Something like:

```
def get_duration(trigger):
    regex = re.compile(...)
    # ... hard work ...
    return timediff.seconds + 86400 * timediff.days
```

And then:

```
def at(bot, trigger):
    # ... 

    try:
        duration = get_duration(trigger)
    except:
        bot.say('Error...')
        return

    if duration < 0:
        duration += 86400
        duration += 86400
    create_reminder(bot, trigger, duration, message, timezone)
    if timezone:
        create_reminder(bot, trigger, duration, message, timezone)
    else:
        create_reminder(bot, trigger, duration, message, 'UTC')
```

Of course, if you don't have time or interest in working on this PR again, I'll take for myself, so we don't lost all this hard work.

**Comment by Exirel** [comment]
I did the thing in #1590, closing here. Feel free to comment/review the new PR, help is still and always be welcome.

**Pull request closed without merging**